packages/data-store/src/utils/digitalCredential/MappingUtils.ts

```diff
@@ -0,0 +1,112 @@
+import { AddDigitalCredentialArgs } from '../../types/digitalCredential/IAbstractDigitalCredentialStore'
+import { DigitalCredentialEntity } from '../../entities/digitalCredential/DigitalCredentialEntity'
+import {
+  CredentialMapper,
+  decodeSdJwtVc,
+  DocumentFormat,
+  IVerifiableCredential,
+  IVerifiablePresentation,
+  SdJwtDecodedVerifiableCredentialPayload,
+} from '@sphereon/ssi-types'
+import {
+  CredentialDocumentFormat,
+  CredentialType,
+  DigitalCredential,
+  NonPersistedDigitalCredential,
+} from '../../types/digitalCredential/digitalCredential'
+import { computeEntryHash } from '@veramo/utils'
+import { createHash } from 'crypto'
+
+function determineCredentialType(raw: string): CredentialType {
+  if (CredentialMapper.hasProof(raw)) {
+    if (CredentialMapper.isCredential(raw)) {
+      return CredentialType.VC
+    } else if (CredentialMapper.isPresentation(raw)) {
+      return CredentialType.VP
+    }
+  } else {
+    if (CredentialMapper.isCredential(raw)) {
+      return CredentialType.C
+    } else if (CredentialMapper.isPresentation(raw)) {
+      return CredentialType.P
+    }
+  }
+  throw new Error(`Couldn't determine the type of the credential: ${raw}`)
+}
+
+function determineCredentialDocumentFormat(documentFormat: DocumentFormat): CredentialDocumentFormat {
+  switch (documentFormat) {
+    case DocumentFormat.JSONLD:
+      return CredentialDocumentFormat.JSON_LD
+    case DocumentFormat.JWT:
+      return CredentialDocumentFormat.JWT
+    case DocumentFormat.SD_JWT_VC:
+      return CredentialDocumentFormat.SD_JWT
+    default:
+      throw new Error(`Not supported document format: ${documentFormat}`)
+  }
+}
+
+function getExpiresAt(uniformDocument: IVerifiableCredential | IVerifiablePresentation | SdJwtDecodedVerifiableCredentialPayload): Date | undefined {
+  if ('expirationDate' in uniformDocument) {
+    return new Date(uniformDocument.expirationDate)
+  } else if ('exp' in uniformDocument) {
+    return new Date(uniformDocument.exp)
+  }
+  return undefined
+}
+
+function getIssuedAt(uniformDocument: IVerifiableCredential | IVerifiablePresentation | SdJwtDecodedVerifiableCredentialPayload): Date | undefined {
```

Comment:
See remark about new names in VCDM 2

Reply:
for the issuance date, I'm also checking the `validFrom` from VCDM2 here. Altough the wording of the document is quite confusing by itself:
`If validFrom and validUntil are not present, the [verifiable credential](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-credential) validity period is considered valid indefinitely. In such cases, the [verifiable credential](https://www.w3.org/TR/vc-data-model-2.0/#dfn-verifiable-credential) is assumed to be valid from the time the verifiable credential was created.`
which suggests it's different from creation of the verifiable credential. We also have this:
from [here](https://www.w3.org/TR/vc-data-model-2.0/#validity-period)
`... which could be a date and time in the future or in the past.`
(from the changes from version 1):
- Make validFrom optional.
- Rename issuanceDate/expirationDate to validFrom/validUntil.

my suggestion is we move to signatures for this matter and use that instead. because every verifiable claim is signed and we can work better with them. Of course we can have extra fields in the db representing the new changes in the VCDM (for validFrom) but since it's optional in the data model, we can't rely on that for our VCs

Reply:
and by the way this is why I've made the getFirstProof method public to use it in here and get the signature's created date.

Reply:
also changed this to `validFrom`

Reply:
Signing has nothing to do with validFrom (previous Issuance)dates/times